### PR TITLE
fix: callback pdf render error

### PR DESCRIPTION
**Pull request by dorianmiller**
#### What do these changes do/fix?

Problem when a PDF fails to render, the document view is blank. The fix defaults to display the text version.

#### How do you test/verify these changes?

Will add a unit test to load an invalid pdf file. 

#### Are there any breaking changes included in this pull request?

No

### CODE REVIEW

File: examples/discovery-search-app/src/ExampleDocumentProvider.js [hunk 1]
```diff
@@ -10,12 +10,31 @@
  */
 export class ExampleDocumentProvider {
   async provides(document: DocumentProviderProps) {
-    return document.extracted_metadata.filename === 'Art Effects.pdf';
+    // return document.extracted_metadata.filename === 'Art Effects.pdf';
+    console.log('dorian change', document.extracted_metadata.filename);
+    if (document.extracted_metadata.filename.indexOf('.pdf') >= 0) {
```
Comment (noah-eigenfeld) [inline examples/discovery-search-app/src/ExampleDocumentProvider.js:15]: I don't know if this difference would ever matter, but you can also use `document.extracted_metadata.file_type === 'pdf'` as your check here (and wherever else you want to verify the document type). Honestly, since you've worked on the backend of Answers, you might have a better idea of how reliable the filename of a document that gets returned to us is.

File: examples/discovery-search-app/src/ExampleDocumentProvider.js [hunk 2]
```diff
@@ -10,12 +10,31 @@
...
+      // hack to force PDF to render
+      document.extracted_metadata.text_mappings = true;
```
Comment (noah-eigenfeld) [inline examples/discovery-search-app/src/ExampleDocumentProvider.js:17]: This seems like a bad practice. We shouldn't be modifying the object getting passed into this provider, because there might be other components or methods that assume this object is coming straight from the api response. If this started affecting the outcome of a completely different component in the future, it would be very hard to debug. We should try to fix how our code is reacting to bad data, or fix the example data at the example file we're feeding in.

Also, according to the typings we have in tooling, the `text_mappings` variable is either a string or an array of strings, but not a boolean.

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx [hunk 3]
```diff
@@ -56,7 +56,11 @@ export type PdfViewerProps = PdfDisplayProps & {
    */
   setRenderedText?: (info: PdfRenderedText | null) => any;
   /**
-   * URL of hosted PDF worker
+   * Callback any errors on render
+   */
+  setIsPdfRenderError: (isError: boolean) => void;
+  /**
+   * Callback for text layer info
```
Comment (noah-eigenfeld) [inline packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx:63]: Why did you change the description of this variable? Shouldn't it still be `URL of hosted PDF worker`?

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerWithHighlight/PdfViewerWithHighlight.tsx [hunk 4]
```diff
@@ -179,6 +190,29 @@ function useHighlightState({
...
+  const [currentIsPdfError, setCurrentIsPdfError] = useState(isPdfRenderError);
+  console.log('useIsPfdError', currentIsPdfError, callbackIsPdfError);
+
+  // Process error message
+  const _previouserrMsgFromPdf = useRef(isPdfRenderError);
+  useEffect(() => {
+    console.log('_previouserrMsgFromPdf', _previouserrMsgFromPdf);
+    console.log('currentErrMsgFromPdf  ', currentIsPdfError);
+    console.log('localErrMsgFromPdf    ', isPdfRenderError);
+    callbackIsPdfError?.(currentIsPdfError);
```
Comment (noah-eigenfeld) [inline packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerWithHighlight/PdfViewerWithHighlight.tsx:210]: What is this line doing?

**Review by noah-eigenfeld — COMMENTED** [review]
Some initial comments and questions

File: examples/discovery-search-app/src/ExampleDocumentProvider.js [hunk 5]
```diff
@@ -10,12 +10,31 @@
...
+      console.log('dorian change, yes pdf');
```
[the remark this answers is not in the document]
Comment (dorianmiller) [reply examples/discovery-search-app/src/ExampleDocumentProvider.js:18]: Thanks, I removed it.

File: examples/discovery-search-app/src/ExampleDocumentProvider.js (hunk 1 above)
Comment (dorianmiller) [reply examples/discovery-search-app/src/ExampleDocumentProvider.js]: I restored the original version. This was for local testing.

File: examples/discovery-search-app/src/ExampleDocumentProvider.js (hunk 2 above)
Comment (dorianmiller) [reply examples/discovery-search-app/src/ExampleDocumentProvider.js]: Good design point. Original file is restored.

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx (hunk 3 above)
Comment (dorianmiller) [reply packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx]: Thanks for catching my copy-and-paste. I restored the original description.

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerWithHighlight/PdfViewerWithHighlight.tsx (hunk 4 above)
Comment (dorianmiller) [reply packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerWithHighlight/PdfViewerWithHighlight.tsx:205]: Calling `callbackIsPdfError` is updating the state and passing the state change from PDFViewer (where error happened) to DocumentPreview (logic changes display from pdf to html/text). PdfViewerWithHighlight is an intermediate component.

**Review by noah-eigenfeld — APPROVED** [review]
